string ades_name
---
KeyValPair[] preconditions
